moviebot/nlu/text_processing.py

```diff
@@ -37,25 +38,24 @@ def overlaps(self, other) -> bool:
         Returns:
             bool: True if there is overlap.
         """
-        return (self.start <= other.start
-                and self.end > other.start) or (other.start <= self.start
-                                                and other.end > self.start)
+        return (self.start <= other.start and self.end > other.start) or (
+            other.start <= self.start and other.end > self.start
+        )
 
     def __eq__(self, other):
-        return (self.start, self.end, self.text, self.lemma) == (
-            other.start,
-            other.end,
-            other.text,
-            other.lemma,
-        )
+        s_dict = vars(self)
+        o_dict = vars(other)
+        return all((s_dict.get(k) == v for k, v in o_dict.items())) and type(
```

Comment:
Nit: compare the types first since it's a simpler check so it fails faster in case they are different.

Reply:
Done.